db/model/Gdoc/GdocFactory.ts

```diff
@@ -1,154 +1,556 @@
+        { ids: ids }
+    )
+    const groupedTags = groupBy(tags, "gdocId")
+    const enrichedRows = rows.map((row) => {
+        return {
+            ...parsePostsGdocsRow(row),
+            tags: groupedTags[row.id] ? groupedTags[row.id] : null,
+        } satisfies OwidGdocBaseInterface
+    })
+    const gdocs = await Promise.all(
+        enrichedRows.map(async (row) => loadGdocFromGdocBase(knex, row))
+    )
+    return gdocs as GdocPost[]
+}
 
-        const gdoc = match(type)
-            .with(
-                P.union(
-                    OwidGdocType.Article,
-                    OwidGdocType.LinearTopicPage,
-                    OwidGdocType.TopicPage,
-                    OwidGdocType.Fragment,
-                    OwidGdocType.AboutPage
-                ),
-                () => GdocPost.create(base)
-            )
-            .with(OwidGdocType.DataInsight, () => GdocDataInsight.create(base))
-            .with(OwidGdocType.Homepage, () => GdocHomepage.create(base))
-            .with(OwidGdocType.Author, () => GdocAuthor.create(base))
-            .exhaustive()
-
-        if (contentSource === GdocsContentSource.Gdocs) {
-            await gdoc.fetchAndEnrichGdoc()
+export async function loadPublishedGdocAuthors(
+    knex: KnexReadonlyTransaction
+): Promise<GdocAuthor[]> {
+    const rows = await knexRaw<DbRawPostGdoc>(
+        knex,
+        `-- sql
+            SELECT *
+            FROM posts_gdocs
+            WHERE published = 1
+            AND content ->> '$.type' IN (:types)`,
+        {
+            types: [OwidGdocType.Author],
         }
+    )
+
+    const enrichedRows = rows.map((row) => {
+        return {
+            ...parsePostsGdocsRow(row),
+        } satisfies OwidGdocBaseInterface
+    })
+    const gdocs = await Promise.all(
+        enrichedRows.map(async (row) => loadGdocFromGdocBase(knex, row))
+    )
+    return gdocs as GdocAuthor[]
+}
+
+export async function getAndLoadListedGdocPosts(
+    knex: KnexReadonlyTransaction
+): Promise<GdocPost[]> {
+    // TODO: Check if we shouldn't also restrict the types of gdocs here
+    const rows = await knexRaw<DbRawPostGdoc>(
+        knex,
+        `-- sql
+            SELECT *
+            FROM posts_gdocs
+            WHERE published = 1
+            publicationContext: ${OwidGdocPublicationContext.listed}
+            publishedAt <= NOW()
+            ORDER BY publishedAt DESC`,
+        {}
+    )
+    const ids = rows.map((row) => row.id)
+    const tags = await knexRaw<DbPlainTag>(
+        knex,
+        `-- sql
+                SELECT gt.gdocId as gdocId, tags.*
+                FROM tags
+                JOIN posts_gdocs_x_tags gt ON gt.tagId = tags.id
+                WHERE gt.gdocId in (:ids)`,
+        { ids: ids }
+    )
+    const groupedTags = groupBy(tags, "gdocId")
+    const enrichedRows = rows.map((row) => {
+        return {
+            ...parsePostsGdocsRow(row),
+            tags: groupedTags[row.id] ? groupedTags[row.id] : null,
```

Comment:
jsyk you can use an or operator here

```
undefined || null // null
null || undefined // undefined
```

I think either is fine, in fact maybe the ternary is even clearer ⭐